iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/MergingVariantManager.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.iidm.mergingview;
+
+import com.powsybl.iidm.network.*;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+/**
+ * @author Thomas Adam <tadam at silicom.fr>
+ */
+class MergingVariantManager implements VariantManager {
+
+    private final MergingViewIndex index;
+
+    private boolean allowVariantMultiThreadAccess = false;
+
+    private String workingVariantId = VariantManagerConstants.INITIAL_VARIANT_ID;
+
+    MergingVariantManager(final MergingViewIndex index) {
+        this.index = Objects.requireNonNull(index, "merging view index is null");
+    }
+
+    private Stream<VariantManager> getVariantManagerStream() {
+        return index.getNetworkStream().map(Network::getVariantManager);
+    }
+
+    @Override
+    public Collection<String> getVariantIds() {
+        final VariantManager vm = getVariantManagerStream().findFirst().orElse(null);
```

Comment:
You should probably throw an exception instead of returning a null: no variant manager means you don't have any network...

Reply:
I will fix it to throw an exception

Reply:
Use orElseThrow instead